src/features/presence/PresenceAvatarList.tsx

```diff
@@ -0,0 +1,180 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import noop from 'lodash/noop';
+import { FormattedMessage, injectIntl, WrappedComponentProps } from 'react-intl';
+// @ts-ignore flow import
+import messages from './messages';
+// @ts-ignore flow import
+import PresenceAvatar from './PresenceAvatar';
+import Tooltip, { TooltipPosition } from '../../components/tooltip';
+// @ts-ignore flow import
+import { determineInteractionMessage } from './utils/presenceUtils';
+import './PresenceAvatarList.scss';
+
+type Collaborator = {
+    avatarUrl?: string;
+    id: string;
+    isActive: boolean;
+    interactedAt: number;
+    interactionType: string;
+    name: string;
+    profileUrl: string;
+};
+
+type Props = {
+    avatarAttributes?: React.HTMLAttributes<HTMLDivElement>;
+    className?: string;
+    collaborators: Array<Collaborator>;
+    containerAttributes?: React.HTMLAttributes<HTMLDivElement>;
+    hideTooltips?: boolean;
+    maxAdditionalCollaborators?: number;
+    maxDisplayedAvatars?: number;
+    onAvatarMouseEnter?: (id: string) => void;
+    onAvatarMouseLeave?: () => void;
+} & WrappedComponentProps;
+
+class PresenceAvatarList extends React.Component<Props> {
+    static defaultProps = {
+        maxAdditionalCollaborators: 99,
+        maxDisplayedAvatars: 3,
+        onAvatarMouseEnter: noop,
+        onAvatarMouseLeave: noop,
+    };
+
+    constructor(props: Props) {
+        super(props);
+
+        this.showTooltip = this.showTooltip.bind(this);
+        this.hideTooltip = this.hideTooltip.bind(this);
+    }
+
+    state = {
+        activeTooltip: null,
+    };
+
+    hideTooltip = (): void => {
+        const { onAvatarMouseLeave } = this.props;
+        this.setState({
+            activeTooltip: null,
+        });
+        if (onAvatarMouseLeave) {
+            onAvatarMouseLeave();
+        }
+    };
+
+    getRelativeTime = (interactedAt: number): string => {
+        const { intl } = this.props;
+
+        if (intl.formatRelativeTime) {
+            return intl.formatRelativeTime(interactedAt - Date.now());
+        }
+        // @ts-ignore: react-intl v2 backwards compatibility
+        return intl.formatRelative(interactedAt);
+    };
+
+    renderTimestampMessage = (interactedAt: number, interactionType: string, isActive: boolean): JSX.Element | null => {
```

Comment:
Can/should we inline and/or break these `render*` helpers into separate components while we're at it?